### PR TITLE
Extraction of code for reuse at CDC  

### PR DESCRIPTION
**Pull request by azexcy**
Related #22500.

Changes proposed in this pull request:
  - Extraction of code for reuse at CDC
  - Adjust database parameter at SocketSinkImporterConnector 

---

Before committing this PR, I'm sure that I have checked the following options:
- [ ] My code follows the [code of conduct](https://shardingsphere.apache.org/community/en/involved/conduct/code/) of this project.
- [ ] I have self-reviewed the commit code.
- [ ] I have (or in comment I request) added corresponding labels for the pull request.
- [ ] I have passed maven check locally : `./mvnw clean install -B -T1C -Dmaven.javadoc.skip -Dmaven.jacoco.skip -e`.
- [ ] I have made corresponding changes to the documentation.
- [ ] I have added corresponding unit tests for my changes.

### CODE REVIEW

File: kernel/data-pipeline/cdc/core/src/test/java/org/apache/shardingsphere/data/pipeline/cdc/util/CDCSchemaTableUtilTest.java [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.mockito.Mockito.mock;
+
+public class CDCSchemaTableUtilTest {
```
Comment (sandynz) [inline kernel/data-pipeline/cdc/core/src/test/java/org/apache/shardingsphere/data/pipeline/cdc/util/CDCSchemaTableUtilTest.java:41]: Class should be `final`

File: kernel/data-pipeline/cdc/core/src/test/java/org/apache/shardingsphere/data/pipeline/cdc/util/CDCSchemaTableUtilTest.java [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+    
+    @Test
+    public void assertParseSchemaTableExpression() {
+        Map<String, ShardingSphereSchema> schemas = new HashMap<>();
+        schemas.put("test", mockSchema());
+        schemas.put("public", mockSchema());
```
Comment (sandynz) [inline kernel/data-pipeline/cdc/core/src/test/java/org/apache/shardingsphere/data/pipeline/cdc/util/CDCSchemaTableUtilTest.java:47]: Could we put different schema for different schema names? It's confusing when schema is the same for test and public schema

File: kernel/data-pipeline/cdc/core/src/test/java/org/apache/shardingsphere/data/pipeline/cdc/util/CDCSchemaTableUtilTest.java [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+        Map<String, ShardingSphereTable> tables = new HashMap<>();
+        tables.put("t_order", mock(ShardingSphereTable.class));
+        tables.put("t_order_item", mock(ShardingSphereTable.class));
+        return new ShardingSphereSchema(tables, Collections.emptyMap());
+    }
+    
+    @Test
+    public void assertParseTableExpression() {
+        Map<String, ShardingSphereSchema> schemas = new HashMap<>();
+        schemas.put("sharding_db", mockSchema());
```
Comment (sandynz) [inline kernel/data-pipeline/cdc/core/src/test/java/org/apache/shardingsphere/data/pipeline/cdc/util/CDCSchemaTableUtilTest.java:78]: What type of `schemas`, it's `sharding_db` here, but it's `test` and `public` in `assertParseSchemaTableExpression`

File: kernel/data-pipeline/cdc/core/src/test/java/org/apache/shardingsphere/data/pipeline/cdc/util/DataRecordResultConvertUtilTest.java [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+        dataRecord.addColumn(new Column("BigInteger", BigInteger.ONE, false, true));
+        dataRecord.addColumn(new Column("BigDecimal", BigDecimal.valueOf(123), false, false));
+        dataRecord.addColumn(new Column("Long", Long.MAX_VALUE, false, false));
+        dataRecord.addColumn(new Column("Integer", Integer.MAX_VALUE, false, false));
+        dataRecord.addColumn(new Column("LocalTime", LocalTime.now(), false, false));
+        Blob mockBlob = mock(Blob.class);
+        when(mockBlob.getBytes(anyLong(), anyInt())).thenReturn(new byte[0]);
+        dataRecord.addColumn(new Column("Blob", mockBlob, false, false));
+        Clob mockClob = mock(Clob.class);
+        when(mockClob.getSubString(anyLong(), anyInt())).thenReturn("");
```
Comment (sandynz) [inline kernel/data-pipeline/cdc/core/src/test/java/org/apache/shardingsphere/data/pipeline/cdc/util/DataRecordResultConvertUtilTest.java:61]: Could we return more meaningful values for `mockBlob` and `mockClob`?
e.g. new byte[]{-1, 0, 1}

Comment (sandynz) [reply kernel/data-pipeline/cdc/core/src/test/java/org/apache/shardingsphere/data/pipeline/cdc/util/DataRecordResultConvertUtilTest.java:61]: `mockBlob` could be `mockedBlob`, and also `mockClob`

File: kernel/data-pipeline/cdc/core/src/test/java/org/apache/shardingsphere/data/pipeline/cdc/util/DataRecordResultConvertUtilTest.java [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+        dataRecord.addColumn(new Column("Timestamp", new Timestamp(System.currentTimeMillis()), false, false));
+        dataRecord.setTableName("t_order");
+        dataRecord.setType("INSERT");
+        Record originRecord = DataRecordResultConvertUtil.convertDataRecordToRecord("test", null, dataRecord);
+        TypeRegistry registry = TypeRegistry.newBuilder().add(CDCResponseProtocol.getDescriptor().getFile().getMessageTypes()).add(WrappersProto.getDescriptor().getMessageTypes())
+                .add(TimestampProto.getDescriptor().getMessageTypes()).build();
+        String print = JsonFormat.printer().usingTypeRegistry(registry).print(originRecord);
+        Builder newRecord = Record.newBuilder();
+        JsonFormat.parser().usingTypeRegistry(registry).merge(print, newRecord);
+        assertEquals(newRecord.build(), originRecord);
```
Comment (sandynz) [inline kernel/data-pipeline/cdc/core/src/test/java/org/apache/shardingsphere/data/pipeline/cdc/util/DataRecordResultConvertUtilTest.java:72]: `newRecord` could be `actualRecord`; `originRecord` could be `expectedRecord`

File: kernel/data-pipeline/cdc/core/src/test/java/org/apache/shardingsphere/data/pipeline/cdc/util/DataRecordResultConvertUtilTest.java [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+public final class DataRecordResultConvertUtilTest {
+    
+    @Test
+    public void assertConvertDataRecordToRecord() throws InvalidProtocolBufferException, SQLException {
+        DataRecord dataRecord = new DataRecord(new IntegerPrimaryKeyPosition(0, 1), 2);
+        dataRecord.addColumn(new Column("BigInteger", BigInteger.ONE, false, true));
+        dataRecord.addColumn(new Column("BigDecimal", BigDecimal.valueOf(123), false, false));
+        dataRecord.addColumn(new Column("Long", Long.MAX_VALUE, false, false));
+        dataRecord.addColumn(new Column("Integer", Integer.MAX_VALUE, false, false));
+        dataRecord.addColumn(new Column("LocalTime", LocalTime.now(), false, false));
```
Comment (sandynz) [inline kernel/data-pipeline/cdc/core/src/test/java/org/apache/shardingsphere/data/pipeline/cdc/util/DataRecordResultConvertUtilTest.java:56]: It's better to put real column name, but not `BigInteger` etc

**Review by sandynz — CHANGES_REQUESTED** [review]

File: kernel/data-pipeline/cdc/core/src/test/java/org/apache/shardingsphere/data/pipeline/cdc/util/CDCSchemaTableUtilTest.java (hunk 3 above)
Comment (azexcy) [reply kernel/data-pipeline/cdc/core/src/test/java/org/apache/shardingsphere/data/pipeline/cdc/util/CDCSchemaTableUtilTest.java:78]: Need to make database name and schema name equals of MySQL, so here is `sharding_db`;

**Review by sandynz — APPROVED** [review]

**Pull request merged**